api/v1beta1/manilaapi_types.go

```diff
@@ -130,7 +130,8 @@ func init() {
 	SchemeBuilder.Register(&ManilaAPI{}, &ManilaAPIList{})
 }
 
-// IsReady - returns true if ManilaAPI is reconciled successfully
+// IsReady - returns true if ManilaAPI is reconciled successfully and is ready to
+// serve requests
 func (instance ManilaAPI) IsReady() bool {
-	return instance.Status.Conditions.IsTrue(condition.ReadyCondition)
+	return instance.Status.ReadyCount == *instance.Spec.Replicas
```

Comment:
Now I have a more clear idea of how the whole conditions related semantics work. and I think this patch can be valid, but only in combination with other changes where the conditions are re-init and re-evaluated for updates.
@abays the `instance.Status.ReadyCount > 0`, as per the doc [1], mentions that a Deployment can be considered Ready when `ReadyCount` is > 0.
This is what we evaluate in the controller of each service (-api, -scheduler, -share).
However, to mark a component as Ready, we should look at the underlying conditions, but also we should match the desired `Replicas`, otherwise there's a problem and the ReadyCondition can't be set.
The status of each component should also influence the top-level Status, but that wasn't part of this patch.
I think as a follow up, but mainly to meet the updates Conditions criteria, we're going to rework this part to make sure that the `isReady` conditions has the right meaning for a given service (or top-level), but keeping consistency with what was agreed in the document [1].
We'll follow up on this, but I'm not going to re-open this patch until cinder and glance are in good shape and we have a well established pattern.  

[1] https://github.com/openstack-k8s-operators/docs/blob/main/conditions.md